IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -387,42 +442,111 @@ - (BOOL)removeAccountsWithKey:(MSIDCacheKey *)key
 #pragma mark - Credentials
 
 // Write a credential to the macOS keychain cache.
-- (BOOL)saveToken:(__unused MSIDCredentialCacheItem *)credential
-              key:(__unused MSIDCacheKey *)key
-       serializer:(__unused id<MSIDCacheItemSerializing>)serializer
-          context:(__unused id<MSIDRequestContext>)context
-            error:(__unused NSError **)error
+- (BOOL)saveToken:(MSIDCredentialCacheItem *)credential
+              key:(MSIDCacheKey *)key
+       serializer:(id<MSIDCacheItemSerializing>)serializer
+          context:(id<MSIDRequestContext>)context
+            error:(NSError **)error
 {
     assert(credential);
     assert(serializer);
+    
     [self updateLastModifiedForCredential:credential context:context];
-    NSData *itemData = [serializer serializeCredentialCacheItem:credential];
+    NSString *credentialKey = [NSString stringWithFormat:@"%@%@%@", key.account, keyDelimiter, key.service];
+    
+    if (key.isShared)
+    {
+        [self.sharedStorageItem storeCredential:credential forKey:credentialKey];
+        return [self saveStorageItem:self.sharedStorageItem key:key serializer:serializer context:context error:error];
+    }
+    else
+    {
+        [self.appStorageItem storeCredential:credential forKey:credentialKey];
+        return [self saveStorageItem:self.appStorageItem key:key serializer:serializer context:context error:error];
+    }
+}
 
+- (BOOL)saveStorageItem:(MSIDMacCredentialStorageItem *)storageItem
+                    key:(MSIDCacheKey *)key
+             serializer:(id<MSIDCacheItemSerializing>)serializer
+                context:(id<MSIDRequestContext>)context
+                  error:(NSError **)error
+{
+    assert(storageItem);
+    MSIDMacCredentialStorageItem *savedStorageItem = [self storageItemWithKey:key serializer:serializer context:context error:error];
+    
+    if (savedStorageItem)
+    {
+        [storageItem mergeStorageItem:savedStorageItem];
+    }
+    
+    NSData *itemData = [serializer serializeCredentialStorageItem:storageItem];
+    
     if (!itemData)
     {
         if (error)
         {
-            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Failed to serialize account item.", nil, nil, nil, context.correlationId, nil);
-            MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Failed to serialize account item.");
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Failed to serialize token.", nil, nil, nil, context.correlationId, nil);
+            MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Failed to serialize token.");
```

Comment:
it might be helpful if we use `credential storage item` rather than `token` in the logging :)

Reply:
Sure.:)